frontend/src/stores/courses.js

```diff
@@ -163,8 +163,10 @@ class courseStore {
             this.selectCourse(null);
             this.checkAvailability();
         } catch (error) {
-            console.error('Cannot reserve course', error);
-            return false;
+            console.log('Cannot reserve course ', error.message);
+            this.rootStore.userStore.checkAuthenticationStatusOnStart();
```

Comment:
why do we need an authentication check here?

Reply:
`checkAuthenticationStatusOnStart()` actually sends a `GET` request to `/users/me` and update user data on frontend accordingly. Full code is here 
```
    checkAuthenticationStatusOnStart = async () => {
        try {
            const userData = await checkLoginStatus();
            this.setCredentials(userData);
        } catch (error) {
            console.error(error);
        }
    };
```

```
     export const checkLoginStatus = () => myFetch(`/users/me`);
```